src/uu/truncate/src/truncate.rs

```diff
@@ -27,78 +28,100 @@ enum TruncateMode {
+    format!("{0} [OPTION]... [FILE]...", executable!())
+}
+
+fn get_long_usage() -> String {
+    format!(
+        "
+    SIZE is an integer with an optional prefix and optional unit.
+    The available units (K, M, G, T, P, E, Z, and Y) use the following format:
+        'KB' =>           1000 (kilobytes)
+        'K'  =>           1024 (kibibytes)
+        'MB' =>      1000*1000 (megabytes)
+        'M'  =>      1024*1024 (mebibytes)
+        'GB' => 1000*1000*1000 (gigabytes)
+        'G'  => 1024*1024*1024 (gibibytes)
+    SIZE may also be prefixed by one of the following to adjust the size of each
+    file based on its current size:
+        '+'  => extend by
+        '-'  => reduce by
+        '<'  => at most
+        '>'  => at least
+        '/'  => round down to multiple of
+        '%'  => round up to multiple of"
+    )
+}
+
 pub fn uumain(args: impl uucore::Args) -> i32 {
+    /*
     let args = args.collect_str();
 
-    let mut opts = getopts::Options::new();
-
-    opts.optflag("c", "no-create", "do not create files that do not exist");
-    opts.optflag(
-        "o",
-        "io-blocks",
-        "treat SIZE as the number of I/O blocks of the file rather than bytes (NOT IMPLEMENTED)",
-    );
-    opts.optopt(
-        "r",
-        "reference",
-        "base the size of each file on the size of RFILE",
-        "RFILE",
-    );
-    opts.optopt("s", "size", "set or adjust the size of each file according to SIZE, which is in bytes unless --io-blocks is specified", "SIZE");
-    opts.optflag("h", "help", "display this help and exit");
-    opts.optflag("V", "version", "output version information and exit");
-
-    let matches = match opts.parse(&args[1..]) {
-        Ok(m) => m,
-        Err(f) => crash!(1, "{}", f),
-    };
+    let mut opts = getopts::Options::new();*/
+    let usage = get_usage();
+    let long_usage = get_long_usage();
+
+    let matches = App::new(executable!())
+        .version(VERSION)
+        .about(ABOUT)
+        .usage(&usage[..])
+        .after_help(&long_usage[..])
+        .arg(
+            Arg::with_name(OPT_NO_CREATE)
+            .short("c")
+            .long(OPT_NO_CREATE)
+            .help("do not create files that do not exist")
+        )
+        .arg(
+            Arg::with_name(OPT_IO_BLOCKS)
+            .short("o")
+            .long(OPT_IO_BLOCKS)
+            .help("treat SIZE as the number of I/O blocks of the file rather than bytes (NOT IMPLEMENTED)")
+        )
+        .arg(
+            Arg::with_name(OPT_REFERENCE)
+            .short("r")
+            .long(OPT_REFERENCE)
+            .help("base the size of each file on the size of RFILE")
+            .value_name("RFILE")
+        )
+        .arg(
+            Arg::with_name(OPT_SIZE)
+            .short("s")
+            .long("size")
+            .help("set or adjust the size of each file according to SIZE, which is in bytes unless --io-blocks is specified")
+            .value_name("SIZE")
+        )
```

Comment:
Unless you see some logical order, I'd alphabetize these by name (`OPT_IO_BLOCKS`, `OPT_NO_CREATE`, `OPT_REFERENCE`, `OPT_SIZE`).